tools/types-generator/cpp_generator.py

```diff
@@ -248,7 +248,13 @@ def print_peer_to_app_component(self, targetFile, namespace):
                 targetFile.write("{}{}(&src->{}, &{});\n".format(gIndent * 2, get_peer_to_app_fn_for_type(self.type), self.name, tmpVarName))
                 targetFile.write("{}{}\n".format(gIndent, "}"))
             else:
-                targetFile.write("{}strcpy(&dst->{}[0], src->{}().c_str());\n".format(gIndent, self.name, self.name))
+                srtlength = int(''.join(self.array[1:-1]))
+                targetFile.write("{}size_t {}_length = strnlen(src->{}().c_str(), {});\n".format(gIndent, self.name, self.name, srtlength))
+                targetFile.write("{}{} {}_length >= {} {}\n".format(gIndent, "if (", self.name, srtlength, ") {"))
```

Comment:
We are checking for "greater than or equal" to but the feedback is "greater than". Can you confirm the intent?